sematic/config/settings.py

```diff
@@ -176,6 +176,13 @@ def get_plugin_setting(
         plugin_settings = {}
 
     if var not in plugin_settings:
+        # _apply_env_var_overrides only applies overrides
+        # to settings present in the settings file
+        # This ensures env var overrides are still applied
+        # if the setting was not set in the file
+        if var.value in os.environ:
```

Comment:
Curious, since the enum name and value are pretty much the same, why do we prefer using `var.value` over `var.name`?

Not a merge blocking question.

Reply:
`var.name` is the variable name we use in the code, `var.value` is the string that is used in the settings file. Right now they are indeed identical. They could diverge at some point.